### ISSUE TITLE
Templates do not have internet access after using new NetVM for sys-firewall and installing sys-whonix-14

### ISSUE THREAD

**Issue by straussmaus** [QubesOS/qubes-issues#4453]
### Qubes OS version:

R4.0

### Affected component(s): 

debian-9 template
fedora-26 template

---

### Steps to reproduce the behavior:

1.  Do a fresh install of Qubes R4
2.  Create a new sys-net-MAC NetVM based on this: https://www.qubes-os.org/doc/anonymizing-your-mac-address/
3.  Set sys-firewall to use sys-net-MAC and delete sys-net (original)
4. Try to update debian9 and fedora26 templates

### Expected behavior:

The qube should open an xterm, fetch packages, and install them

### Actual behavior:

An xterm opens, it attempts to download packages, but it can't.  In the fedora-26 template I see this error:

> Error: Failed to synchronize cache for repo 'updates'
> Done.
> Press Enter to shutdown the template, or Ctrl-C to just close this window

### General notes:

ping does not work from the templates.  I think it can not update only because the templates can not get internet.  I think most likely problem is  sys-net being replaced with  another NetVM?  

If I have left something out please tell me and I will try to give answers.  I see every reason why this is a valid issue based on https://www.qubes-os.org/doc/reporting-bugs/.  (But I do not want to prejudice this because I do not see that the team may decide what is an "issue."  By definition that is the users job.  Anyway in this case they overlap completely based on https://www.qubes-os.org/doc/reporting-bugs/.)

This is output from `sudo systemctl status qubes-updates-proxy.service`
```● qubes-updates-proxy.service - Qubes updates proxy (tinyproxy)
   Loaded: loaded (/usr/lib/systemd/system/qubes-updates-proxy.service; enabled; vendor preset: enabled)
   Active: inactive (dead)
Condition: start condition failed at Mon 2018-10-29 11:25:45 GMT; 2min 25s ago
           ├─ ConditionPathExists=|/var/run/qubes-service/qubes-yum-proxy was not met
           └─ ConditionPathExists=|/var/run/qubes-service/qubes-updates-proxy was not met
```

The files `/var/run/qubes-service/qubes-updates-proxy` and /var/run/qubes-service/qubes-yum-proxy` do not exist.  Is it supposed to?

This is output from `sudo journalctl --boot | grep qubes-updates-proxy` on fedora-26:
```
Oct 29 10:52:53 fedora-26 kernel: audit: type=1130 audit(1540810373.479:73): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@0-127.0.0.1:8082-127.0.0.1:49238 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 29 10:52:53 fedora-26 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@0-127.0.0.1:8082-127.0.0.1:49238 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 29 10:52:53 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@0-127.0.0.1:8082-127.0.0.1:49238.service: Main process exited, code=exited, status=126/n/a
Oct 29 10:52:53 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@0-127.0.0.1:8082-127.0.0.1:49238.service: Unit entered failed state.
Oct 29 10:52:53 fedora-26 audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@0-127.0.0.1:8082-127.0.0.1:49238 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Oct 29 10:52:53 fedora-26 kernel: audit: type=1131 audit(1540810373.622:74): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@0-127.0.0.1:8082-127.0.0.1:49238 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Oct 29 10:52:53 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@0-127.0.0.1:8082-127.0.0.1:49238.service: Failed with result 'exit-code'.
Oct 29 11:19:58 fedora-26 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@1-127.0.0.1:8082-127.0.0.1:49240 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 29 11:19:58 fedora-26 audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@1-127.0.0.1:8082-127.0.0.1:49240 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Oct 29 11:19:58 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@1-127.0.0.1:8082-127.0.0.1:49240.service: Main process exited, code=exited, status=126/n/a
Oct 29 11:19:58 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@1-127.0.0.1:8082-127.0.0.1:49240.service: Unit entered failed state.
Oct 29 11:19:58 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@1-127.0.0.1:8082-127.0.0.1:49240.service: Failed with result 'exit-code'.
Oct 29 11:24:18 fedora-26 kernel: audit: type=1130 audit(1540812258.376:110): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@2-127.0.0.1:8082-127.0.0.1:49242 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 29 11:24:18 fedora-26 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@2-127.0.0.1:8082-127.0.0.1:49242 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Oct 29 11:24:18 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@2-127.0.0.1:8082-127.0.0.1:49242.service: Main process exited, code=exited, status=126/n/a
Oct 29 11:24:18 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@2-127.0.0.1:8082-127.0.0.1:49242.service: Unit entered failed state.
Oct 29 11:24:18 fedora-26 audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@2-127.0.0.1:8082-127.0.0.1:49242 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Oct 29 11:24:18 fedora-26 kernel: audit: type=1131 audit(1540812258.471:111): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=qubes-updates-proxy-forwarder@2-127.0.0.1:8082-127.0.0.1:49242 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
Oct 29 11:24:18 fedora-26 systemd[1]: qubes-updates-proxy-forwarder@2-127.0.0.1:8082-127.0.0.1:49242.service: Failed with result 'exit-code'.
Oct 29 11:25:45 fedora-26 sudo[2117]:     user : TTY=pts/0 ; PWD=/home/user ; USER=root ; COMMAND=/bin/systemctl restart qubes-updates-proxy
Oct 29 11:25:54 fedora-26 sudo[2126]:     user : TTY=pts/0 ; PWD=/home/user ; USER=root ; COMMAND=/bin/systemctl status qubes-updates-proxy.sevice
Oct 29 11:25:56 fedora-26 sudo[2135]:     user : TTY=pts/0 ; PWD=/home/user ; USER=root ; COMMAND=/bin/systemctl status qubes-updates-proxy.service

```

---

### Related issues:

I do *not* think it is related to these and I have looked at them:
https://github.com/QubesOS/qubes-issues/issues/3135
https://github.com/QubesOS/qubes-issues/issues/1352
https://github.com/QubesOS/qubes-issues/issues/3557
https://github.com/QubesOS/qubes-issues/issues/3737

**Comment by unman** [comment]
@straussmaus TemplateVMs do not have internet access by design, so it's no surprise that you cannot ping from them.
If you check [the documentation](https://www.qubes-os.org/doc/software-update-vm/) you'll see that updating is done using qubes-rpc using a Proxy - the proxy is set in `/etc/qubes-rpc/policy/qubes.UpdatesProxy.`

I cant reproduce your problem.

Please check the policy file to see which qube you are using for the update Proxy, and then confirm that you have internet access from _that_ qube.

**Comment by straussmaus** [comment]
@unman  I did consult this page: https://www.qubes-os.org/doc/software-update-vm/

On that page there is no discussion that the template does not have internet access.  The opposite is true.  From that page: 

> Install/update software as usual (e.g. using dnf, or the dedicated GUI application).

Because dnf requires internet this page is saying the equal to the fact that the template has internet.

But I accept that you say ping should not work.  How then is a good way I can test and debug the problem.

> I cant reproduce your problem.

Did you install Qubes from nothing since I wrote the ticket?  And then follow the MAC change instructions?  And then update whonix-14?  All within one hour?  This is the situation where there is a problem.

On my system prior there was no problem with the templates.  I could use dnf or sudo apt-get update any time and I could also do "Update Qube" option.  Now I can't and what happens is that I get the error message in my post. 

Because this is a fresh install and I followed the MAC documents and then the whonix-14 documents this is a bug in the code or an error/bug in the instructions.

And we have a solution!  You said: 

> updating is done using qubes-rpc using a Proxy - the proxy is set in /etc/qubes-rpc/policy/qubes.UpdatesProxy.

I saw this in the documents but there is a bug in the documents.  The documents (and also your explanation above) does not say that this file is *on dom0*.  It is important to be precise @unman.  I originally looked for this file in the template and it did not exist so I assumed it was old documentation that does not apply to the new release.  

*Solution to the problem is to edit `/etc/qubes-rpc/policy/qubes.UpdatesProxy` in dom0 to have the name of the new NetVM.*

So the correct action for the devs is to make two changes:

- Edit https://www.qubes-os.org/doc/software-update-vm/ to specify that this file is on dom0.

- Edit https://www.qubes-os.org/doc/anonymizing-your-mac-address/ to specify that `/etc/qubes-rpc/policy/qubes.UpdatesProxy` on dom0 must be updated if you follow the instructions on that page but use a new NetVM with a name that is not sys-net.  (Really this is not *necessary* because the page does not tell you to use a new NetVM so the first one is the only real bug.)

A project is only as good as the written instructions.  I am happy that I was able to help you find this bug and also suggest the second thing as an improvement.

**Comment by andrewdavidwong** [comment]
The documentation is a community effort. Please help us improve it by submitting a pull request:

https://www.qubes-os.org/doc/doc-guidelines/

**Comment by unman** [comment]
@straussmaus 
I'm glad you were able to solve your problem.
It should be clear now that dnf in Templates does _not_ require internet access - that's exactly what the qubes-rpc service is for.

The written instructions are a collaborative exercise, and help is always welcome. When you close this issue, please help by improving the documentation - there is an "Edit this page" button on each page.

**Comment by unman** [comment]
@straussmaus 
`Did you install Qubes from nothing since I wrote the ticket? And then follow the MAC change instructions? And then update whonix-14? All within one hour?`

Whonix-14 wasn't mentioned in your report.
As to the rest, I have stored images of base installs and default templates, so spinning up a clean install is straightforward and quick. The Macchanger instructions are simple and take little time. (I use a loaded caching proxy (apt-cacher-ng) for template updates, so even if I *had* needed to update to match your description of the issue it would have taken little time.)
